src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs

```diff
@@ -50,6 +50,12 @@ public static void Initialize()
         // as you can't make a constructor partial.
         private RuntimeEventSource(int _) { }
 
+        [Event(2, Level = EventLevel.Informational)]
```

Comment:
For ease of use, it might be best to assign this a keyword. 

When you got to enable this, it'll create the counters. They won't get turned on unless you specify `EventCounterIntervalSec` as a filter parameter, though. That being said, we don't want to have the presence of the filter string be the switch for enabling counters or not.

You can do this by adding a nested class like this:
```csharp
public static class Keywords
{
	public const EventKeyword AppContext = (EventKeyword)0x1;
}
```

Then you can assign it to the event by updating the `EventAttribute` to have the `EventKeyword` parameter.

This way a user can turn on _just_ this event without having to turn on all the counters.

CC @brianrob to weigh in on the experience of this.

Reply:
Sorry, I'm not understanding this at all. Are you saying that we should add an additional registration key for just grabbing the AppContext switches?

Reply:
Let me try to re-state @josalem's position and see if I get it right.  I think that @josalem is saying that right now, everything in this EventSource gets turned on all at once because there is no ability to filter.  The counters get created but not actually enabled because a filter string is required to turn them on, but I think he's saying that in the future, he'd like for this not to be the case, and so wants to future proof this with a keyword.  Keywords are one of the constructs that let you filter, and so I believe he's just proposing adding a keyword to allow this to be turned on and off independently of the counters. @josalem did I get that right?

I have another question on this - is this where we would want this and other future managed code events to live, when their functionality is in CoreLib, or do we think that the event should live somewhere like `FrameworkEventSource`?

Reply:
I should also add that I think this event is a great idea, and I will absolutely turn this on by default in PerfView.

Reply:
I think this is *a* reasonable place for the AppContext switch events to live -- I'm not sure I'd be bold enough to take a stronger position than that.

Reply:
Fair enough.  I was aiming that question more at @josalem.  :)

Reply:
Your assessment is spot-on, Brian.

I think it is fine for now. If we add a keyword for the AppContext switches, we can at least turn on just those events. Not specifying `EventCounterIntervalSec` means we can turn on the world without enabling the counters, but it's not ideal as an enabling mechanism.